msal/src/test/java/com/microsoft/identity/client/robolectric/tests/AcquireTokenBaseTest.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.client.robolectric.tests;
+
+import android.app.Activity;
+import android.content.Context;
+
+import androidx.test.core.app.ApplicationProvider;
+
+import com.microsoft.identity.client.IPublicClientApplication;
+import com.microsoft.identity.client.PublicClientApplication;
+import com.microsoft.identity.client.exception.MsalException;
+
+import org.mockito.Mockito;
+
+import java.io.File;
+
+public abstract class AcquireTokenBaseTest {
+
+    private static final String CONFIG_FILE_PATH = "src/test/res/raw/aad_test_config.json";
+
+    abstract void makeAcquireTokenCall(final IPublicClientApplication publicClientApplication,
+                                       final Activity activity) throws InterruptedException;
+
+
+    private Activity getActivity(final Context context) {
+        final Activity mockedActivity = Mockito.mock(Activity.class);
+        Mockito.when(mockedActivity.getApplicationContext()).thenReturn(context);
```

Comment:
Curious why mocking with `Mockito` here, I thought `Roboelectric` already provides API for that ?

Reply:
Provides API for what?

Reply:
API to mock an activity/ context ? Surprised  Roboelectric is not already providing a basic shadow to mock an activity

Reply:
Robolectric gives us a real context (mocked context but mocked really well). Once we have the context, we can do pretty much anything. We can also use Robolectric's `setupActivity` method to setup a specific activity. I wouldn't think of that as mocking though as the Activity returned is like a real Activity (we can click button and stuff). 

Since we do not have any Activities in our library we did not need to use that. Specifically for these tests without UI Automation, we just need a mocked Activity as we don't need to do anything with it.